src/pseudotransient.jl

```diff
@@ -114,12 +114,16 @@ end
 function perform_step!(cache::PseudoTransientCache{true})
     @unpack u, u_prev, fu1, f, p, alg, J, linsolve, du, alpha, tc_storage = cache
     jacobian!!(J, cache)
-    J_new = J - (1 / alpha) * I
+    if J isa SciMLBase.AbstractSciMLOperator
+        J = J - (1 / alpha) * I
+    else
+        J .= J - (1 / alpha) * I
```

Comment:
This still allocates. Use the same flow as https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/src/derivative_utils.jl#L494-L503